bson-kotlinx/build.gradle.kts

```diff
@@ -45,6 +45,7 @@ dependencies {
 
     implementation(platform("org.jetbrains.kotlinx:kotlinx-serialization-bom:1.5.0"))
     implementation("org.jetbrains.kotlinx:kotlinx-serialization-core")
+    implementation("org.jetbrains.kotlinx:kotlinx-datetime:0.4.0")
```

Comment:
This version is compatible with Kotlin 1.5 onward.

Reply:
We can't take a hard dependency on this library - so this will need refactoring to make it optional.  #1459 adds a check for kotlinx-json support, you could use that style to add an if statement for the `defaultSerializersModule`